### PR TITLE
Add Vagrantfile for running tests in development

### PR DESCRIPTION
**Pull request by dwick**
👓 @ckwang8128 

cc @spladug for visibility

Guess, check, google, repeat on puppet so feel free to be hard on me!

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     'python3-posix-ipc',
+    ]:
+      ensure  => installed,
+      require => [
+        Exec['reddit-repo-add'], # Add the reddit ppa
+        Exec['update-apt'],      # The system update needs to run
+      ],
+  } ->
+
+  package {
```
Comment (spladug) [inline manifests/init.pp]: Why have these guys come from pip?

File: manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Installation of initial dependencies and baseplate project
+# in a development environment.
+# Please add additional dependencies and env setup code as needed.
+class baseplate {
+  $project_path = '/home/vagrant/baseplate'
+
+  group { 'puppet':
```
Comment (spladug) [inline manifests/init.pp]: What's this for?

**Comment by spladug** [comment]
:nail_care: just a couple tiny questions. looks great. would you mind running `puppet-lint` on the puppetfile for style please?

File: manifests/init.pp (hunk 2 above)
Comment (ckwang8128) [inline manifests/init.pp]: this was copied from the manifest I linked as an example in the cookiecutter template https://github.com/reddit/reddit-cookiecutter-pyramid/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/manifests/init.pp

I think that was erroneously copied from an example I used as I was learning how to write puppet manifests properly. it shouldn't be needed here (or there).

File: manifests/init.pp (hunk 1 above)
Comment (ckwang8128) [inline manifests/init.pp]: just checked, there are deb packages available for all of these for python2 and python3 so it should be safe to add them to the above stanza!

Comment (dwick) [inline manifests/init.pp]: i was looking at the travis config for deps, wasn't sure why these were pip there either :)

**Comment by dwick** [comment]
💇

**Comment by spladug** [comment]
:fish:

**Pull request merged**